pricegraph/src/encoding.rs

```diff
@@ -29,6 +29,41 @@ pub struct TokenPair {
     pub sell: TokenId,
 }
 
+impl TokenPair {
+    pub fn inverse(self) -> Self {
+        TokenPair {
+            buy: self.sell,
+            sell: self.buy,
+        }
+    }
+
+    pub fn into_unbounded_range(self) -> TokenPairRange {
+        self.into_range(None)
+    }
+
+    pub fn into_range(self, hops: Option<usize>) -> TokenPairRange {
+        TokenPairRange { pair: self, hops }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+pub struct TokenPairRange {
```

Comment:
ubernit: Not 100% happy with the naming of this, but I also can't think of anything better.

Reply:
How about `Itinerary` (consists of start/end some general meta data). Or to make it not too far fetched `BoundedTokenPair` or `TokenPairInfo`

Reply:
`Itinerary` sounds cool, but a little abstract. Lets just leave it as is.